core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java

```diff
@@ -0,0 +1,126 @@
+package alluxio.master.file.activesync;
+
+import alluxio.AlluxioURI;
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.exception.InvalidPathException;
+import alluxio.heartbeat.HeartbeatContext;
+import alluxio.heartbeat.HeartbeatThread;
+import alluxio.master.file.ActiveSyncer;
+import alluxio.master.file.FileSystemMaster;
+import alluxio.master.file.meta.MountTable;
+import alluxio.resource.CloseableResource;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.util.UnderFileSystemUtils;
+import alluxio.util.io.PathUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+public class ActiveSyncManager {
+  private static final Logger LOG = LoggerFactory.getLogger(ActiveSyncManager.class);
+  private final MountTable mMountTable;
+
+  private final List<AlluxioURI> mSyncPathList;
+  // a map which maps each UFS rootPath to a thread polling that UFS
+  private final Map<String, Future<?>> mPollerMap;
+  //  a map which maps each UFS rootPath to a list of paths being actively synced on that thread
+  private final Map<String, List<AlluxioURI>> mFilterMap;
+
+  private FileSystemMaster mFileSystemMaster;
+
+  public ActiveSyncManager(MountTable mountTable,
+      FileSystemMaster fileSystemMaster) {
+    mMountTable = mountTable;
+    mPollerMap = new ConcurrentHashMap<>();
+    mFilterMap = new ConcurrentHashMap<>();
+    mSyncPathList = new CopyOnWriteArrayList<>();
+    mFileSystemMaster = fileSystemMaster;
+  }
+
+  public boolean isActivelySynced(AlluxioURI path) throws InvalidPathException {
+    for (AlluxioURI syncedPath : mSyncPathList) {
+      if (PathUtils.hasPrefix(path.getPath(), syncedPath.getPath())) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public boolean addSyncPoint(AlluxioURI syncPoint) throws InvalidPathException {
+    LOG.info("adding syncPoint {}", syncPoint.getPath());
+    if (!isActivelySynced(syncPoint)) {
+      MountTable.Resolution resolution = mMountTable.resolve(syncPoint);
+      AlluxioURI ufsUri = resolution.getUri();
+      String rootPath = ufsUri.getRootPath();
```

Comment:
I'm not sure what the distinction is between rootpath, sync point, and mount point are. (I only truly know what a mount point is)